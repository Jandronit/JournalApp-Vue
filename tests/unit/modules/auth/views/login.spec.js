import { shallowMount } from "@vue/test-utils";
import { useRouter } from "vue-router";
import Login from "@/modules/auth/views/Login.vue";
import createVuexStore from "../../../../mocks/mock-store";
jest.mock("vue-router", () => ({
  useRouter: jest.fn(),
}));

describe("Tests Login Component", () => {
  const store = createVuexStore({
    status: "not-authenticated",
    user: null,
    idToken: null,
    refreshToken: null,
  });
  beforeEach(() => jest.clearAllMocks());

  test("should match snapshot", () => {
    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store],
        mocks: {
          $router: useRouter(),
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
