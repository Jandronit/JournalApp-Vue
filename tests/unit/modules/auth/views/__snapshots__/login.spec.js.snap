// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Login Component should match snapshot 1`] = `
<span class="login100-form-title p-b-41"> Log In </span>
<form class="login100-form validate-form p-b-33 p-t-5">
  <div class="wrap-input100 validate-input" data-validate="Enter username"><input class="input100" type="email" placeholder="Email" autocomplete="username" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Enter password"><input class="input100" type="password" placeholder="Password" autocomplete="current-password" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="container-login100-form-btn m-t-32"><button type="submit" class="login100-form-btn">Login</button></div>
  <div class="container-login100-form-btn m-t-32">
    <router-link to="[object Object]">You do not have an account?</router-link>
  </div>
</form>
`;
